python/taichi/lang/ast/ast_transformer.py

```diff
@@ -427,6 +427,15 @@ def build_call_if_is_builtin(ctx, node, args, keywords):
             id(pow): pow,
             id(operator.matmul): matrix_ops.matmul,
         }
+        if id(func) == id(len):
+            if len(args) != 1:
+                raise TaichiSyntaxError("len() takes exactly one argument.")
+            if isinstance(args[0], Expr) and args[0].ptr.is_tensor():
+                node.ptr = args[0].get_shape()[0]
+            else:
+                node.ptr = len(*args, **keywords)
```

Comment:
What about only handling the special case (one argument with `TensorType`) and let other cases fall into the default `build_Call`? Then we don't have to report the `SyntaxError` above either.